tools/ld-chroma-decoder/main.cpp

```diff
@@ -449,12 +474,16 @@ int main(int argc, char *argv[])
         combConfig.adaptive = false;
         decoder.reset(new NtscDecoder(combConfig));
     } else if (decoderName == "mono") {
-        decoder.reset(new MonoDecoder);
+        decoder.reset(new MonoDecoder(combConfig));
     } else {
         qCritical() << "Unknown decoder " << decoderName;
         return -1;
     }
 
+    decoder->config.pix_fmt = combConfig.outputYCbCr ?
```

Comment:
As above, this shouldn't be necessary. Have a pixelFormat field in Decoder::Configuration using an enumerated type, and set it above with the other configuration that gets passed to the Decoder constructors - directly modifying it here isn't very nice.

Reply:
Yes.  I felt mean doing it.